data/helpers.d/user

```diff
@@ -52,7 +53,12 @@ ynh_system_user_create () {
 		else
 			local user_home_dir="--no-create-home"
 		fi
-		sudo useradd $user_home_dir --system --user-group $1 --shell /usr/sbin/nologin || ynh_die "Unable to create $1 system account"
+		if [ $# -ge 3 ]; then	# If we want a shell for the user
```

Comment:
Hm I don't know what's the best design for this :sweat_smile: That's a small thing but here you could run `ynh_system_user_create foo /bin/false` and it would actually use the default login shell because the third argument is taken as a boolean argument. Though it's difficult to design an elegant solution ... eitheir explicitly gives the shell to use (e.g. `/bin/bash`), or use literal values like `True / False` or `DEFAULT_SHELL / NO_LOGIN` ?

Or we can keep it like this, but at least an example of usage would be good to clarify this :stuck_out_tongue: 

(Otherwise that looks good to me though not tested)

Reply:
...that's why I was seeking your opinion, because I couldn't find out something completely satisfying myself :sweat_smile:
Providing the shell as parameter seems overkill and prevents from using the default shell (unless you allow using the `default` parameter).
Adding an example is definitely a good start!

Reply:
Hum, and here using a specific shell means also you have to specify a home directory.
Sometimes, I'm asking myself if we shouldn't switch from positional arguments to getopts arguments for the helpers.

Reply:
Provided we keep simple enough helpers (which is a good practice!), I recommend to keep positional arguments.

Reply:
I found a few hours today, I made a helper for using getopts easily.
https://github.com/YunoHost-Apps/Experimental_helpers/pull/21

@JimboJoe, what do you think about this helper ? Don't you think that using this helper would permit to use "getopts" arguments without the complexity of getopts

Reply:
In the case of this helper, what would you propose as parameters definition?
Moreover, any idea as how to manage helper API compatibility with existing apps...?

Reply:
Anyway, as the helper should currently be tested before be merged in the core. This PR shouldn't wait for it.
So let's use positional arguments here, and we will make a new PR when the new helper getopts will be ready.